File: javascript/difference-of-squares/difference-of-squares.js
export class Squares {
  constructor(n) {
    this.n = n;
  }

  get sumOfSquares() {
    let sum = 0;
    for (let i = 1; i <= this.n; i++) {
      sum += i ** 2;
    }
    return sum;
  }

  get squareOfSum() {
    return (this.n * (this.n + 1) / 2) ** 2;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}

File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(number):
    return int(number * (number + 1) / 2) ** 2


def sum_of_squares(number):
    output = 0
    for i in range(1, number + 1):
        output += i ** 2
    return output


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)
